Source/VideoProgressView.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+
+class VideoProgressView: UIView {
+
+    static let Height = CGFloat(55.0)
+    private static let ProgressBarXMargin = CGFloat(65.0)
+    private static let ProgressBarYMargin = CGFloat(23.0)
+    private static let ProgressBarHeight = CGFloat(6.0)
+
+    private static let TextLabelWidth = CGFloat(36.0)
+    private static let TextLabelHeight = CGFloat(18.0)
+    private static let TextLabelMargin = CGFloat(18.0)
+
+    var duration = 0.0 {
+        didSet {
+            if self.duration != oldValue {
+              self.durationTimeLabel.text = self.timeStringForSeconds(self.duration)
+            }
+        }
+    }
+    var currentTime = 0.0 {
+        didSet {
+            self.currentTimeLabel.text = self.timeStringForSeconds(self.currentTime)
+            self.setFrameForProgressBar()
+        }
+    }
+
+    var progressPercentage : Double {
+        guard self.currentTime != 0.0 && self.duration != 0.0 else {
+            return 0.0
+        }
+
+        return currentTime/duration
+    }
+
+    var widthForBar : CGFloat {
+        return self.bounds.width - (2 * VideoProgressView.ProgressBarXMargin)
+    }
+
+    var widthForProgressBar : CGFloat {
+        return widthForBar * CGFloat(progressPercentage)
+    }
+
+    lazy var maskBarForRoundedCorners: UIView = {
+        let maskView = UIView()
+        maskView.backgroundColor = UIColor.clearColor()
+        maskView.layer.cornerRadius = ProgressBarHeight/2
+        maskView.clipsToBounds = true
+        maskView.layer.masksToBounds = true;
+        return maskView
```

Comment:
I like to keep the returns separated by a newline, I've noticed Apple does this too. There are a few places where this change might be required.

https://github.com/apple/example-package-deckofplayingcards/blob/master/Sources/Deck.swift#L41-L42